### PR TITLE
Convert metadata columns from integer to columns to ensure that SDV works properly

### LINKED ISSUE #1933: Metadata detection crashes when the column names are integers (`AttributeError: 'int' object has no attribute 'lower'`)
### Environment Details
* SDV version: 1.12.0 (latest)

### Error Description
If my datafame's column names are represented with integers (instead of with strings), then the metadata detection crashes with a `TypeError`. 

### Steps to reproduce

```python
import pandas as pd
import numpy as np

from sdv.metadata import SingleTableMetadata

# generate a dataframe with integer column names
num_rows = 100
num_cols = 20
values = {i+1: np.random.randint(0, 100, size=num_rows) for i in range(num_cols)}
data = pd.DataFrame(values)

metadata = SingleTableMetadata()
metadata.detect_from_dataframe(data=df)
```
```
AttributeError: 'int' object has no attribute 'lower'
```
[stack_trace.txt](https://github.com/sdv-dev/SDV/files/15030588/stack_trace.txt)

### Additional Context
**When would you have a DataFrame with integer column names?** One use case is if I have a CSV that doesn't have any column names listed. If I tell pandas that there is no header in my data, then it will create column names with index values `0`, `1`, ...

```python
df = pd.read_csv('my_data.csv', header=None)
```

### Workaround
A simple workaround would be to rename the columns using strings instead of integers.

### PR DESCRIPTION
**Pull request by lajohn4747**
resolves #1933 
CU-86b04u2ae

Metadata columns need to be strings as much of SDV metadata detection assumes  all columns are of `string` type

### CODE REVIEW

**Comment by sdv-team** [comment]
Task linked: [CU-86b04u2ae SDV - Metadata detection crashes when the column names are integers (`TypeError: unsupported operand type(s)`) #1933](https://app.clickup.com/t/86b04u2ae)

File: sdv/multi_table/hma.py [hunk 1]
```diff
@@ -114,7 +114,7 @@ def _estimate_columns_traversal(cls, metadata, table_name,
             columns_per_table[table_name] += \
                 cls._get_num_extended_columns(
                     metadata, child_name, table_name, columns_per_table, distributions
-                )
+            )
```
Comment (amontanez24) [inline sdv/multi_table/hma.py:117]: why move this? The bracket should line up with the statement it closes right?

**Review by amontanez24 — COMMENTED** [review]
Can we add an integration test for both the single and multi table case?

File: sdv/metadata/multi_table.py [hunk 2]
```diff
@@ -524,6 +524,9 @@ def _detect_relationships(self):
     def detect_table_from_dataframe(self, table_name, data):
         """Detect the metadata for a table from a dataframe.
 
+        This method automatically detects the ``sdtypes`` for the given ``pandas.DataFrame``,
+        for a specified table. All data columns are converted to strings.
```
Comment (amontanez24) [inline sdv/metadata/multi_table.py:528]: we should change this to say column names so they don't think we're converting the columns themselves

**Review by amontanez24 — APPROVED** [review]

File: tests/integration/single_table/test_base.py [hunk 3]
```diff
@@ -831,3 +831,20 @@ def test_sample_not_fitted(synthesizer):
...
+        1: [1, 2, 3],
+        2: [4, 5, 6],
+        3: ['a', 'b', 'c'],
+    })
+    metadata = SingleTableMetadata()
+    metadata.detect_from_dataframe(data)
+    instance = synthesizer_class(metadata)
+    instance.fit(data)
+    sample = instance.sample(5)
+    assert sample.columns.tolist() == data.columns.tolist()
```
Comment (R-Palazzo) [inline tests/integration/single_table/test_base.py:850]: and `# Assert` and `# Run` before

**Review by R-Palazzo — APPROVED** [review]
LGTM!

**Pull request merged**